packages/app/src/State/Login.ts

```diff
@@ -315,12 +325,16 @@ const LoginSlice = createSlice({
       window.localStorage.setItem(PrivateKeyItem, action.payload);
       state.publicKey = secp.utils.bytesToHex(secp.schnorr.getPublicKey(action.payload));
     },
-    setGeneratedPrivateKey: (state, action: PayloadAction<HexKey>) => {
+    setGeneratedPrivateKey: (state, action: PayloadAction<SetGeneratedKeyPayload>) => {
       state.loggedOut = false;
       state.newUserKey = true;
-      state.privateKey = action.payload;
-      window.localStorage.setItem(PrivateKeyItem, action.payload);
-      state.publicKey = secp.utils.bytesToHex(secp.schnorr.getPublicKey(action.payload));
+      state.privateKey = action.payload.key;
+      state.generatedEntropy = action.payload.entropy;
+      window.localStorage.setItem(PrivateKeyItem, action.payload.key);
```

Comment:
If you don't store the entropy in local storage then on-reload you will lose the mnemonic, also its a bit confusing that in some places its called mnemonic and here it's called entropy.

Isn't the mnemonic just a 256bit value the same as a bare private key?

Reply:
No sorry you're right, i see they are actually different things, i thought you were storing the mnemonic always.

And in regards to not keeping the value, you done it perfectly, i see its cleared when exiting the discover flow